design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,357 @@
+Azure resource that has made the request has a Conjur identity.
+
+```yaml
+"xms_mirid": "/subscriptions/<subscription_id>/resourcegroups/<resource_group>/providers/Microsoft.ContainerInstance/containerGroups/<assigned_identity>"
+```
+
+| Host annotation                                                      | xms_mirid              |
+|----------------------------------------------------------------------|------------------------|
+| `authn-azure/subscription-id`                                        | subscription_id        |
+| `authn-azure/resource-group`                                         | resource_group         |
+| `authn-azure/user-assigned-identity` or `system-assigned-identity`   | assigned_identity      |
+
+### Backwards compatibility
+
+None because this is a new feature.
+
+### Performance
+
+Azure authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+
+- Conjur
+
+- DAP
+
+## Test Plan
+
+TBT
+
+### Integration tests
+
+#### Functionality
+
+This suite of tests assumes that the Azure authenticator has been configured correctly
+
+|    | Given                                                                                        | When                                                     | Then                                                                                          | Manual / UT / Integration | Status     |
+|----|----------------------------------------------------------------------------------------------|----------------------------------------------------------|-----------------------------------------------------------------------------------------------|---------------------------|------------|
+| 1  | *Vanilla flow* - a Conjur identity exists for Azure VM and added to privileged layer         | VM sends authentication request with a valid Azure token | Conjur access token is returned in body and action is logged                                  | Integration               |            |
+| 2  | A Conjur Host identity for Azure VM, added to privileged layer and has permissions on secret | VM sends request with valid Azure token to fetch secret  | Secret is successfully returned and action is logged                                          | Integration               |            |
+| 3  | No Conjur Host identity exists for Azure VM                                                  | VM sends authentication request with a valid Azure token | 403 Forbidden response / RoleNotFound in error body is returned and action is logged          | Integration               |            |
+| 4  | Conjur identity exists for Azure VM but doesn't have permissions on secret                   | VM sends request with valid Azure token to fetch secret  | 401 Unauthorized response is returned and action is logged                                    | Integration               |            |
+
+#### Error handling
```

Comment:
Im not sure all the errors here are 500.  According to this [document](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) .
Some of them may be 501,503.